// hume/dist/serialization/resources/expressionMeasurement/resources/stream/types/StreamModelPredictions.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamModelPredictions = void 0;
const core = __importStar(require("../../../../../../core"));
const StreamModelPredictionsJobDetails_1 = require("./StreamModelPredictionsJobDetails");
const StreamModelPredictionsBurst_1 = require("./StreamModelPredictionsBurst");
const StreamModelPredictionsFace_1 = require("./StreamModelPredictionsFace");
const StreamModelPredictionsFacemesh_1 = require("./StreamModelPredictionsFacemesh");
const StreamModelPredictionsLanguage_1 = require("./StreamModelPredictionsLanguage");
const StreamModelPredictionsProsody_1 = require("./StreamModelPredictionsProsody");
exports.StreamModelPredictions = core.serialization.object({
    payloadId: core.serialization.property("payload_id", core.serialization.string().optional()),
    jobDetails: core.serialization.property("job_details", StreamModelPredictionsJobDetails_1.StreamModelPredictionsJobDetails.optional()),
    burst: StreamModelPredictionsBurst_1.StreamModelPredictionsBurst.optional(),
    face: StreamModelPredictionsFace_1.StreamModelPredictionsFace.optional(),
    facemesh: StreamModelPredictionsFacemesh_1.StreamModelPredictionsFacemesh.optional(),
    language: StreamModelPredictionsLanguage_1.StreamModelPredictionsLanguage.optional(),
    prosody: StreamModelPredictionsProsody_1.StreamModelPredictionsProsody.optional(),
});
