// hume/serialization/resources/empathicVoice/resources/configs/client/requests/PostedConfigVersion.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PostedConfigVersion = void 0;
const core = __importStar(require("../../../../../../../core"));
const PostedConfigPromptSpec_1 = require("../../../../types/PostedConfigPromptSpec");
const PostedVoice_1 = require("../../../../types/PostedVoice");
const PostedLanguageModel_1 = require("../../../../types/PostedLanguageModel");
const PostedEllmModel_1 = require("../../../../types/PostedEllmModel");
const PostedUserDefinedToolSpec_1 = require("../../../../types/PostedUserDefinedToolSpec");
const PostedBuiltinTool_1 = require("../../../../types/PostedBuiltinTool");
const PostedEventMessageSpecs_1 = require("../../../../types/PostedEventMessageSpecs");
const PostedTimeoutSpecs_1 = require("../../../../types/PostedTimeoutSpecs");
const PostedWebhookSpec_1 = require("../../../../types/PostedWebhookSpec");
exports.PostedConfigVersion = core.serialization.object({
    eviVersion: core.serialization.property("evi_version", core.serialization.string()),
    versionDescription: core.serialization.property("version_description", core.serialization.string().optional()),
    prompt: PostedConfigPromptSpec_1.PostedConfigPromptSpec.optional(),
    voice: PostedVoice_1.PostedVoice.optional(),
    languageModel: core.serialization.property("language_model", PostedLanguageModel_1.PostedLanguageModel.optional()),
    ellmModel: core.serialization.property("ellm_model", PostedEllmModel_1.PostedEllmModel.optional()),
    tools: core.serialization.list(PostedUserDefinedToolSpec_1.PostedUserDefinedToolSpec.optional()).optional(),
    builtinTools: core.serialization.property("builtin_tools", core.serialization.list(PostedBuiltinTool_1.PostedBuiltinTool.optional()).optional()),
    eventMessages: core.serialization.property("event_messages", PostedEventMessageSpecs_1.PostedEventMessageSpecs.optional()),
    timeouts: PostedTimeoutSpecs_1.PostedTimeoutSpecs.optional(),
    webhooks: core.serialization.list(PostedWebhookSpec_1.PostedWebhookSpec.optional()).optional(),
});
